Assets/AppCenter/Plugins/AppCenterSDK/Push/Android/PushInternal.cs

```diff
@@ -1,67 +1,85 @@
-        private static AndroidJavaClass _unityListener = new AndroidJavaClass("com.microsoft.appcenter.pushdelegate.UnityAppCenterPushDelegate");
-
-        public static void PrepareEventHandlers()
-        {
-            AppCenterBehavior.InitializingServices += Initialize;
-            AppCenterBehavior.InitializedAppCenterAndServices += PostInitialize;
-        }
-
-        private static void Initialize()
-        {
-            _unityListener.CallStatic("setListener", new PushDelegate());
-        }
-
-        public static void StartPush()
-        {
-        }
-
-        private static void PostInitialize()
-        {
-            var instance = _push.CallStatic<AndroidJavaObject>("getInstance");
-            AndroidJavaClass unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
-            AndroidJavaObject activity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity");
-            instance.Call("onActivityResumed", activity);
-        }
-
-        public static AppCenterTask SetEnabledAsync(bool isEnabled)
-        {
-            var future = _push.CallStatic<AndroidJavaObject>("setEnabled", isEnabled);
-            return new AppCenterTask(future);
-        }
-
-        public static AppCenterTask<bool> IsEnabledAsync()
-        {
-            var future = _push.CallStatic<AndroidJavaObject>("isEnabled");
-            return new AppCenterTask<bool>(future);
-        }
-
-        public static void AddNativeType(List<IntPtr> nativeTypes)
-        {
-            nativeTypes.Add(AndroidJNI.FindClass("com/microsoft/appcenter/push/Push"));
-        }
-
-        public static void EnableFirebaseAnalytics()
-        {
-            _push.CallStatic("enableFirebaseAnalytics");
-        }
-
-        internal static void ReplayUnprocessedPushNotifications()
-        {
-            _unityListener.CallStatic("replayPushNotifications");
-        }
-    }
-}
-#endif
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#if UNITY_ANDROID && !UNITY_EDITOR
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AppCenter.Unity.Push.Internal
+{
+    class PushInternal
+    {
+        private static AndroidJavaClass _push = new AndroidJavaClass("com.microsoft.appcenter.push.Push");
+        private static AndroidJavaClass _unityListener = new AndroidJavaClass("com.microsoft.appcenter.pushdelegate.UnityAppCenterPushDelegate");
+        private static bool IsAppCenterStarted = false;
+        private static bool IsWaitingForReplay = false;
+
+        public static void PrepareEventHandlers()
+        {
+            AppCenterBehavior.InitializingServices += Initialize;
+            AppCenterBehavior.InitializedAppCenterAndServices += PostInitialize;
+        }
+
+        private static void Initialize()
+        {
+            _unityListener.CallStatic("setListener", new PushDelegate());
+            IsAppCenterStarted = true;

```

Comment:
Same code is repeated. Maybe you can move those to Shared/Push and call a method from here. Not a big deal though.

Reply:
The idea is to replay it when platform specific listener is initialized, so it looks OK to have this check in platform specific initialize